### PR TITLE
Add support for attachments to the Foundation cross-import overlay.

### PR DESCRIPTION
**Pull request by grynspan**
This PR adds experimental support for attachments to some types in Foundation via the (non-functional) cross-import overlay. @stmontgomery is working on setting up said overlay so that it can actually be used; until then, the changes here are speculative only.

### Checklist:

- [x] Code and documentation should follow the style of the [Style Guide](https://github.com/apple/swift-testing/blob/main/Documentation/StyleGuide.md).
- [x] If public symbols are renamed or modified, DocC references should be updated.

### CODE REVIEW

**Comment by grynspan** [comment]
@swift-ci test

**Comment by grynspan** [comment]
@swift-ci test

**Comment by grynspan** [comment]
@swift-ci test

**Comment by grynspan** [comment]
@swift-ci test

**Comment by grynspan** [comment]
@swift-ci test

**Comment by grynspan** [comment]
@swift-ci test

**Comment by grynspan** [comment]
@swift-ci test

**Comment by grynspan** [comment]
@swift-ci test

File: Sources/Overlays/_Testing_Foundation/Attachments/EncodingFormat.swift [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  case propertyListFormat(_ format: PropertyListSerialization.PropertyListFormat)
+
+  /// The JSON format.
+  case json
+
+  /// The encoding format to use by default.
+  ///
+  /// The specific format this case corresponds to depends on if we are encoding
+  /// an `Encodable` value or an `NSSecureCoding` value.
+  case `default`
```
Comment (briancroom) [inline Sources/Overlays/_Testing_Foundation/Attachments/EncodingFormat.swift:35]: Should this be the first case in the list?

Comment (grynspan) [reply Sources/Overlays/_Testing_Foundation/Attachments/EncodingFormat.swift:35]: I can move it if it bothers you—it's an internal type so it's not going to affect client code.

**Comment by grynspan** [comment]
Closing for now: I want to sort out generic attachments first.

**Pull request closed without merging**